compiler-rt/lib/ubsan/ubsan_type_hash_itanium.cpp

```diff
@@ -207,7 +207,8 @@ struct VtablePrefix {
   std::type_info *TypeInfo;
 };
 VtablePrefix *getVtablePrefix(void *Vtable) {
-  Vtable = ptrauth_strip(Vtable, ptrauth_key_cxx_vtable_pointer);
+  Vtable = reinterpret_cast<void *>(
```

Comment:
what is the purpose of this change?

Reply:
@alanzhao1 I think with typeof it should be redundant as return value would have `void*` type. But please double check, I do not have pac-ret enabled compiler-rt right now in buildable state...

Reply:
Chrome was seeing build failures when building compiler-rt for android - see https://crbug.com/355271781#comment3.

The problem was that if we used the inline asm version of `ptrauth_strip(...)`, it woud return a `unsigned long` that would be assigned to `Vtable`, a variable of type `void *`, causing an incompatible type conversion compile failure.

This PR has been updated to fix the return value of `ptrauth_strip(...)`, but LLVM's style seem to be to explicitly cast the value to the desired type anyways.

Reply:
Ended up removing the `reintepret_cast` here.